src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -828,6 +829,80 @@ public static BinaryOperandsKind GetBinaryOperandsKind(BinaryOperationKind kind)
         }
     }
 
+    public interface IConversion
```

Comment:
I cannot speak for @JohnHamby. My suggestion was to incorporate information about language specific conversion kind into the value returned by 
```
ConversionKind ConversionKind { get; }
``` 

And then add convenience extension methods for ConversionKind: IsImplicit, IsIdentity, etc.